.github/workflows/ci.yaml

```diff
@@ -162,7 +179,7 @@ jobs:
       - run: |
           git status
           git clean -f -d
-          FINCH_DOCKER_COMPAT=1 make test-e2e
+          FINCH_DOCKER_COMPAT=1 REGISTRY=${{ env.REGISTRY }} make test-e2e
```

Comment:
I think we also need to add a guard here. Might want to make the variable reusable earlier on, in its own step, as well.

Something like this should work. Otherwise, the test is going to try to pull from the registry but it won't have creds to make the pull actually work

```
      - name: Set reusable variables
        id: vars
        run: |
          has_creds=${{ github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name }}
          echo "has_creds=$has_creds" >> $GITHUB_OUTPUT
      - name: configure aws credentials
        uses: aws-actions/configure-aws-credentials@v2.2.0
        if: steps.vars.outputs.has_creds == true
        with:
...
     - run: |
          git status
          git clean -f -d
          FINCH_DOCKER_COMPAT=1 REGISTRY=${{ steps.vars.outputs.has_creds == true && env.REGISTRY || '' }} make test-e2e
```